src/hardhat/contracts/Staking/FraxUnifiedFarm_ERC20.sol

```diff
@@ -316,6 +395,43 @@ contract FraxUnifiedFarm_ERC20 is FraxUnifiedFarmTemplate {
         _updateRewardAndBalance(msg.sender, false);
     }
 
+    // Extend the lock period on an existing stake by secs
+    function extendLock(bytes32 kek_id, uint256 arr_idx, uint256 secs) updateRewardAndBalance(msg.sender, true) public returns (bytes32) {
+        require(stakingPaused == false || valid_migrators[msg.sender] == true, "Staking paused or in migration");
+
+        // Get the stake
+        LockedStake memory thisStake = _getStake(msg.sender, kek_id, arr_idx);
+
+        // The start_timestamp is not currently used in weight calcs. Leaving as original lock start.
+        uint256 newStartTimestamp = thisStake.start_timestamp;
+        uint256 newEndingTimestamp = thisStake.ending_timestamp + secs;
+        uint256 newLockDurationSecs = newEndingTimestamp - newStartTimestamp;
+        require(newLockDurationSecs >= lock_time_min, "Minimum stake time not met");
+        require(newLockDurationSecs <= lock_time_for_max_multiplier, "Trying to lock for too long");
+
+        // Question: Are we ok to use the original kek_id, or should it be creating a new one?
+        //bytes32 new_kek_id = keccak256(abi.encodePacked(msg.sender, newStartTimestamp, thisStake.liquidity, _locked_liquidity[msg.sender]));
```

Comment:
should be fine using the og kek_id, so far as i can tell.

Reply:
prob a question for Travis